### PR TITLE
created group

### PR DESCRIPTION
**Pull request by bekasatuboldiev**
created group

### CODE REVIEW

**Review by zamiramanasova — COMMENTED** [review]
viewed

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/GroupController.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/group")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/GroupController.java:21]: add ("api/groups")

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table(name = "groups")
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java:14]: Use Getter and Setter

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class Group {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "group_gen")
+    @SequenceGenerator(name = "group_gen",sequenceName = "group_seq",allocationSize = 1)
+    private Long id;
+    @Column(name = "group_name")
+    private String groupName;
+    @Column(name = "start_of_group")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java:23]: You can use just startDate

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private Date starOfGroup;
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java:24]: startDate

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupRequest.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.Setter;
+import java.util.Date;
+
+@Getter
+@Setter
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+public class GroupRequest {
+
+    private String group_name;
+    private Date star_of_group;
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupRequest.java:16]: startDate

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupResponse.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+@Getter
+@Setter
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@Builder
+public class GroupResponse {
+
+    private Long id;
+    private String group_name;
+    private Date star_of_group;
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupResponse.java:20]: startDate

**Review by ChyngyzIshen — COMMENTED** [review]
comments

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java (hunk 2 above)
Comment (zamiramanasova) [reply src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Group.java]: @ChyngyzIshen  can we use Lombok?

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupRequest.java [hunk 7]
```diff
@@ -13,7 +13,7 @@
 public class GroupRequest {
 
     private String group_name;
-    private Date star_of_group;
+    private Date start_date;
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupRequest.java:16]: use camelCase

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupResponse.java [hunk 8]
```diff
@@ -17,7 +17,7 @@ public class GroupResponse {
 
     private Long id;
     private String group_name;
-    private Date star_of_group;
+    private Date start_date;
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/dto/group/GroupResponse.java:20]: camelCase

**Comment by bekasatuboldiev** [comment]
fixed

**Pull request merged**